src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -569,7 +569,7 @@ class DottyBackendInterface(outputDirectory: AbstractFile)(implicit ctx: Context
     def isConstructor: Boolean = toDenot(sym).isConstructor
     def isAnonymousFunction: Boolean = toDenot(sym).isAnonymousFunction
     def isMethod: Boolean = sym is Flags.Method
-    def isPublic: Boolean =  sym.flags.is(Flags.EmptyFlags, Flags.Private | Flags.Protected)
+    def isPublic: Boolean =  toDenot(sym).isPublic
```

Comment:
I agree, but note that `isPublic` in scalac is defined in the same way: https://github.com/scala/scala/blob/2.12.x/src/reflect/scala/reflect/internal/HasFlags.scala#L117
So I think the backend already takes this into account.